server/src/main/java/dev/slimevr/tracking/trackers/Tracker.kt

```diff
@@ -52,13 +58,44 @@ class Tracker @JvmOverloads constructor(
 
 	var status = TrackerStatus.DISCONNECTED
 		set(value) {
-			if (field != value) {
-				field = value
-				if (!isInternal) {
-					// If the status of a non-internal tracker has changed, inform
-					// the VRServer to recreate the skeleton, as it may need to
-					// assign or un-assign the tracker to a body part
-					vrServer.updateSkeletonModel()
+			if (field == value) return
+			field = value
+			if (!isInternal) {
+				// If the status of a non-internal tracker has changed, inform
+				// the VRServer to recreate the skeleton, as it may need to
+				// assign or un-assign the tracker to a body part
+				vrServer.updateSkeletonModel()
+
+				if (lastErrorStatus == 0u && value == TrackerStatus.ERROR) {
+					reportErrorStatus()
+				} else if (lastErrorStatus != 0u && value != TrackerStatus.ERROR) {
+					vrServer.statusSystem.removeStatus(lastErrorStatus)
+					lastErrorStatus = 0u
+				}
+
+				if (needsReset && lastResetStatus == 0u) {
+					if (value.sendData && trackerPosition != null) {
+						reportRequireReset()
+					}
+				} else if (!value.sendData && lastResetStatus != 0u) {
+					vrServer.statusSystem.removeStatus(lastResetStatus)
+					lastResetStatus = 0u
+				}
+			}
+		}
+
+	var trackerPosition = trackerPosition
+		set(value) {
+			if (field == value) return
+			field = value
+
+			if (!isInternal) {
+				// Check if a full reset is needed
+				if (needsReset && value != null && lastResetStatus == 0u && status.sendData) {
+					reportRequireReset()
+				} else if (value == null && lastResetStatus != 0u) {
+					vrServer.statusSystem.removeStatus(lastResetStatus)
+					lastResetStatus = 0u
```

Comment:
This seems like an incredibly large amount of logic to have in a property setter...

Reply:
any suggestions?